### DISCUSSION TITLE
Really confused by Github Oauth not creating a new user.

### DISCUSSION THREAD

**Discussion by murmuur22** [supabase/supabase#29096] [Questions] [answered]
I am relatively new to supabase and full stack development, and am mostly self taught so I apologize if I am not providing the right information.

edit: I am using Sveltekit, Svelte5, and DrizzleORM.

I setup a basic github OAuth system following this tutorial, https://www.youtube.com/watch?v=lEWghUOta-4&t=76s. In my local server it works perfectly, but once I push to vercel everything breaks. 

In my supabase auth log I am getting this error. 

```JSON
{"component":"api","error":"failed to close prepared statement: ERROR: current transaction is aborted, commands ignored until end of transaction block (SQLSTATE 25P02): ERROR: null value in column \"username\" of relation \"profiles\" violates not-null constraint (SQLSTATE 23502)","level":"error","method":"GET","msg":"500: Database error saving new user","path":"/callback","referer":"https://wicked-earth.vercel.app/","remote_addr":"129.21.86.216","request_id":"8be022f2765843b5-EWR","time":"2024-09-04T18:45:25Z"}
```

At first it makes sense, like ok the issue is in the profiles table that I made... sure. So I checked my script that adds to that table.

Here is my script that adds to the profiles table:

```JS
export const getOrCreateUserProfile = async (locals: App.Locals) => {
  try {
    // get user from supabase
    const { user } = await locals.safeGetSession();

    // log the result of safeGetSession
    log.plain("Result from safeGetSession: " + JSON.stringify(user));

    // return null if no user
    if (!user) {
      log.error("No user found in session");
      return null;
    }

    log.plain(" Hello from getOrCreateUserProfile: " + user.id + " " + user.email);

    // check if profile exists
    const currentProfile = await db.query.profiles.findFirst({
      where: eq(profiles.id, user.id)
    });

    // return profile if exists
    if (currentProfile) {
      log.plain("Profile exists for user: " + user.id);
      return currentProfile;
    }

    // create profile
    const newProfile = await db.insert(profiles).values({
      id: user.id,
      email: user.email,
      username: ""
    }).returning("*");

    log.plain("New profile created for user: " + user.id);
    return newProfile;
  } catch (err: any) {
    log.error("Error in getOrCreateUserProfile: " + err.message);
    throw error(500, "Database error saving new user");
  }
}
```

After doing some extensive logging, I found that this function is exiting at this statement

```JS
    // return null if no user
    if (!user) {
      log.error("No user found in session");
      return null;
    }
```

so does that mean the user is null? I checked my `safeGetSession()` script (which I copied from the supabase docs) to find out.

```JS
event.locals.safeGetSession = async () => {
    log.plain("Attempting to get session from Supabase");
    const {
      data: { session },
      error: sessionError
    } = await event.locals.supabase.auth.getSession()

    if (sessionError) {
      log.error("Error getting session: " + sessionError.message);
    }

    if (!session) {
      log.error("No session found");
      return { session: null, user: null }
    }

    log.plain("Session found: " + JSON.stringify(session));

    const {
      data: { user },
      error: userError,
    } = await event.locals.supabase.auth.getUser()
    if (userError) {
      // JWT validation has failed
      log.error("Error getting user: " + userError.message);
      return { session: null, user: null }
    }
    if (!user) {
      log.error("No user found");
      return { session: null, user: null }
    }

    log.plain("User found: " + JSON.stringify(user));

    return { session, user }
  }

  return resolve(event, {
    filterSerializedResponseHeaders(name) {
      /**
       * Supabase libraries use the `content-range` and `x-supabase-api-version`
       * headers, so we need to tell SvelteKit to pass it through.
       */
      return name === 'content-range' || name === 'x-supabase-api-version'
    },
  })
```

I added more console logs and found that it is exiting here

```JS
if (!session) {
      log.error("No session found");
      return { session: null, user: null }
    }
```

and what is more interesting is the if statement above this that is supposed to output an error message doesn't trigger.... so does that mean there is no error??? than why is there no session?...

I feel like maybe I setup the link between github and supabase incorrectly, but I checked like 12 times and it seems right.

It works on my local server, but not on the vercel one. Idk why... again i'm kinda new so maybe i'm missing something but idk.

But thats where I'm at and I am so confused, i've been on this for like two days.. 

If anyone has any insight it'd be a great help. I really want to understand what is going on.

**Comment by GaryAustin1** [comment, marked as answer]
That error is typical of a trigger function on auth.users doing an operation on profiles.  This is usually part of a user management guide from Supabase.

The specific error seems to be the value in username being inserted already exists and you have unique column for it.

You don't show any postgres functions or triggers here though.

**Reply by murmuur22** [reply]
Oh yeah I think you are right. I didn't even know this was here. I don't remember making it though, does it just come with supabase or did I make it and forget... 

Also I am having an issue removing it, it says that an object depends on it, but I don't have anything else in my project. I even tried deleting my empty profiles table and still can't delete it.

```SQL
begin
  insert into public.profiles (id, username)
  values (new.id, new.raw_user_meta_data ->> 'username');
  return new;
end;
```

**Reply by GaryAustin1** [reply]
It does not come automatically, but many guides have it.

You need to delete the trigger from auth.users in the SQL editor or add CASCADE to the function drop SQL.

**Reply by murmuur22** [reply]
That worked! thanks for the help, idk if I would have found that otherwise lol. Now it works as expected. Sanity restored 😮‍💨
